flocker/control/httpapi.py

```diff
@@ -492,11 +498,15 @@ def create_container_configuration(self, host, name, image, ports=()):
                 external_port=port['external']
             ))
 
+        if environment:
```

Comment:
Better to specifically check `if environment is not None` ie not the placeholder value in the function signature. Other Falsy values should be  used and cause a failure if they happen to be supplied.